commit 1915a6814099f6a5711a039cf7f0ab480b332688 (refs/changes/93/593/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-10-04T13:38:14-07:00 (3 years ago)
    
    joyent/node-cueball#34 ConnectionSet must emit 'removed' as soon as FSM leaves 'idle'

diff --git a/lib/set.js b/lib/set.js
index f81dac1..d5ab716 100644
--- a/lib/set.js
+++ b/lib/set.js
@@ -441,12 +441,15 @@ CueBallConnectionSet.prototype.addConnection = function (key) {
 			return;
 		}
 
-		if (newState === 'closed') {
+		if (newState !== 'idle') {
 			if (self.cs_connections[ckey]) {
 				conn = self.cs_connections[ckey];
 				delete (self.cs_connections[ckey]);
 				self.assertEmit('removed', ckey, conn);
 			}
+		}
+
+		if (newState === 'closed') {
 			if (self.cs_fsms[key]) {
 				var idx = self.cs_fsms[key].indexOf(fsm);
 				self.cs_fsms[key].splice(idx, 1);
diff --git a/package.json b/package.json
index 207a85e..f4bd49d 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "cueball",
-  "version": "1.0.2",
+  "version": "1.0.3",
   "description": "",
   "main": "lib/index.js",
   "dependencies": {
